contracts/MemeBase.sol

```diff
@@ -114,7 +111,10 @@ contract MemeBase is MemeFactory {
             payable(address(this)), false);
 
         // Perform swap
-        return IBalancer(balancerVault).swap(singleSwap, fundManagement, 0, block.timestamp);
+        olasAmount = IBalancer(balancerVault).swap(singleSwap, fundManagement, 0, block.timestamp);
+
+        // Apply slippage protection
```

Comment:
same

Reply:
It is not critical to check this before the own swap. Because in case of an attack we are in the middle of the sandwich. If we check at the end, then the price change due to our own exchange will be included in the slippage. Then it just has to be big. The current code does not contradict the practices. Both options will work well against an attack through taking a flashloan at the beginning. Because the price acceleration in case of an flashloan attack is 25 times.